OpenSim/Moco/MocoFrameOrientationConstraint.h

```diff
@@ -0,0 +1,80 @@
+#ifndef OPENSIM_MOCOFRAMEORIENTATIONCONSTRAINT_H 
+#define OPENSIM_MOCOFRAMEORIENTATIONCONSTRAINT_H 
+/* -------------------------------------------------------------------------- *
+ * OpenSim Moco: MocoFrameOrientationConstraint.h                             *
+ * -------------------------------------------------------------------------- *
+ *                                                                            *
+ * Author(s): Size Zheng                                              		  *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0          *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#include "MocoConstraint.h"
+#include <OpenSim/Moco/osimMoco.h>
+#include "osimMocoDLL.h"
+
+namespace OpenSim {
+
+class OSIMMOCO_API MocoFrameOrientationConstraintPair : public Object {
+    OpenSim_DECLARE_CONCRETE_OBJECT(MocoFrameOrientationConstraintPair, Object);
+
+public:
+    OpenSim_DECLARE_PROPERTY(frame1_path, std::string,
+        "The first model frame path of the pair.");
+    OpenSim_DECLARE_PROPERTY(frame2_path, std::string,
+        "The second model frame path of the pair.");
+    OpenSim_DECLARE_PROPERTY(minimum_angle, double,
+        "The minimum distance apart that the two frame origins can be "
+        "(meters).");
+    OpenSim_DECLARE_PROPERTY(maximum_angle, double,
+        "The maximum distance apart that the two frame origins can be "
+        "(meters).")
+
+    MocoFrameOrientationConstraintPair();
+    MocoFrameOrientationConstraintPair(std::string firstFramePath,
+        std::string secondFramePath, double minimum_angle,
+        double maximum_angle);
+
+private:
+    void constructProperties();
+};
+
+class OSIMMOCO_API MocoFrameOrientationConstraint : public MocoPathConstraint {

```

Comment:
Similar to other constraints and MocoGoals, add a comment describing this constraint here.